recipes-asteroid/qml-asteroid-config/qml-asteroid-config_1.0.bb

```diff
@@ -0,0 +1,31 @@
+SUMMARY = "Generates a machine specific config file for AsteroidOS"
+LICENSE = "GPLv3"
+LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/GPL-3.0;md5=c79ff39f19dfec6d293b95dea7b07891"
+
+PACKAGE_ARCH = "${MACHINE_ARCH}"
+
+python do_generate_config () {
+    machine_vars = ['MACHINE_DISPLAY_FLAT_TIRE', 'MACHINE_DISPLAY_BORDER_GESTURE_WIDTH', 'MACHINE_DISPLAY_ROUND', 'MACHINE_HAS_WLAN', 'MACHINE_HAS_SPEAKER']
+
+    from configparser import ConfigParser
+    config = ConfigParser()
+    config.optionxform=str
+    config.add_section('General')
+    for v in machine_vars:
+        if d.getVar(v) is not None:
+            config.set('General', v[len("MACHINE_"):], str(d.getVar(v)))
+
+    with open(d.getVar('WORKDIR') + "/machine.conf", "w") as machine_conf:
+        config.write(machine_conf)
+}
+
+do_install_append() {
+    if [ -f ../machine.conf ]; then
```

Comment:
Do we want to ignore if this file is missing?
My understanding is that your do_generate_config would always write a file. If the file is missing, this is probably worth breaking the build here instead of silently continuing without a machine config.

Reply:
My initial reason for having this was that `qml-asteroid` doesn't crash when the config file doesn't exist. But you are correct, that `do_generate_config` always creates the `machine.conf`.

I will adjust this so that the build will fail when the `do_generate_config` task fails.